server/src/test/java/org/elasticsearch/action/admin/indices/delete/DeleteIndexBlocksIT.java

```diff
@@ -63,7 +66,23 @@ public void testDeleteIndexOnIndexReadOnlyAllowDeleteSetting() {
         }
     }
 
-    public void testDeleteIndexOnReadOnlyAllowDeleteSetting() {
+    public void testClusterBlockMessageHasIndexName() {
+        try {
+            createIndex("test");
+            ensureGreen("test");
+            Settings settings = Settings.builder().put(IndexMetaData.SETTING_READ_ONLY_ALLOW_DELETE, true).build();
+            client().admin().indices().prepareUpdateSettings("test").setSettings(settings).get();
+            ClusterBlockException e = expectThrows(ClusterBlockException.class, () ->
+                client().prepareIndex().setIndex("test").setType("doc").setId("1").setSource("foo", "bar").get());
+            assertEquals("index [test] blocked by: [FORBIDDEN/12/index read-only / allow delete (api)];", e.getMessage());
+        } finally {
+            Settings settings = Settings.builder().putNull(IndexMetaData.SETTING_READ_ONLY_ALLOW_DELETE).build();
+            assertAcked(client().admin().indices().prepareUpdateSettings("test").setIndicesOptions(IndicesOptions.lenientExpandOpen()).
```

Comment:
Maybe inline the settings and remove indices options.

Reply:
just to be clear like this:

```assertAcked(client().admin().indices().prepareUpdateSettings("test")        .setSettings(Settings.builder().putNull(IndexMetaData.SETTING_READ_ONLY_ALLOW_DELETE).build()).get());```

Reply:
Yes